### PR TITLE
Optionally remove boilerplate in initial setup

### LINKED ISSUE #49: Make Python boilerplate optional in initial setup
This makes the project cleaner for other languages, e.g. R, Julia, etc.

### PR DESCRIPTION
**Pull request by chrisjkuch**
Closes #49 

Provides additional prompt to make boilerplate python module files and folders disappear, with the exception of `__init__.py` to make the module a python module.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -40,6 +41,10 @@ def config_generator(fast=False):
         ],
         [("dependency_file", opt) for opt in cookiecutter_json["dependency_file"]],
         [("pydata_packages", opt) for opt in cookiecutter_json["pydata_packages"]],
+        [
+            ("include_code_scaffold", opt)
+            for opt in cookiecutter_json["include_code_scaffold"]
+        ],
```
Comment (chrisjkuch) [inline tests/conftest.py:47]: This doesn't need to be in the `product`

File: {{ cookiecutter.repo_name }}/Makefile [hunk 2]
```diff
@@ -37,11 +36,13 @@ clean:
 lint:
 	flake8 {{ cookiecutter.module_name }}
 	black --check --config pyproject.toml {{ cookiecutter.module_name }}
+	isort --check --profile black {{ cookiecutter.module_name }}
```
Comment (chrisjkuch) [inline {{ cookiecutter.repo_name }}/Makefile:39]: Let's not do this as part of this PR

**Review by pjbull — APPROVED** [review]
Looks good, thanks @chrisjkuch!

File: hooks/post_gen_project.py [hunk 3]
```diff
@@ -14,12 +15,21 @@
     "isort",
     "pip",
     "python-dotenv",
+    "setuptools",
+    "wheel",
```
Comment (jayqi) [inline hooks/post_gen_project.py:19]: @chrisjkuch why do we need setuptools and wheel? These should be unnecessary for a pyproject.toml-based build (this stuff gets specified in pyproject.toml).

**Pull request merged**